core/modules/wdb/QueryAction.cc

```diff
@@ -164,6 +167,9 @@ QueryAction::Impl::Impl(QueryActionArg const& a)
       _poisoned(false),
       _sendChannel(a.task->sendChannel),
       _user(a.task->user) {
+    char buf[MAX_NAME_LENGTH];
+    gethostname(buf, MAX_NAME_LENGTH);
+    _hostName = buf;
```

Comment:
Shouldn't that constant be POSIX's HOST_NAME_MAX?

And note that if the name is truncated there is no guarantee that a trailing null byte will be written to the array.

http://pubs.opengroup.org/onlinepubs/9699919799/functions/gethostname.html